File: backend/src/main/java/fr/pedalons/dto/admin/CreateDomainAliasRequest.java
package fr.pedalons.dto.admin;

import fr.pedalons.dto.validation.ValidateSchema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Request to create a domain alias (dedicated hostname pinned to a team)")
@ValidateSchema
public record CreateDomainAliasRequest(
    @NotBlank @Size(max = 250) @Schema(description = "Alias hostname", required = true)
        String hostname,
    @NotBlank
        @Schema(
            description = "Slug of the team to pin (must belong to the domain)",
            required = true)
        String teamSlug,
    @NotBlank @Size(max = 250) @Schema(description = "Alias display name", required = true)
        String name,
    @NotBlank @Size(max = 500) @Schema(description = "Base URL for the alias", required = true)
        String baseUrl,
    @Size(max = 1000)
        @Schema(
            description =
                "Android app SHA-256 certificate fingerprints for passkey origin verification"
                    + " (comma-separated, colon-hex format)")
        String androidFingerprints) {}

File: frontend/src/api/dto/createDomainAliasRequest.ts
/**
 * Request to create a domain alias (dedicated hostname pinned to a team)
 */
export interface CreateDomainAliasRequest {
  /**
   * Alias hostname
   * @maxLength 250
   * @pattern \S
   */
  hostname: string
  /**
   * Slug of the team to pin (must belong to the domain)
   * @pattern \S
   */
  teamSlug: string
  /**
   * Alias display name
   * @maxLength 250
   * @pattern \S
   */
  name: string
  /**
   * Base URL for the alias
   * @maxLength 500
   * @pattern \S
   */
  baseUrl: string
  /**
   * Android app SHA-256 certificate fingerprints for passkey origin verification (comma-separated, colon-hex format)
   * @maxLength 1000
   */
  androidFingerprints?: string
}
